### PR TITLE
Transfer fee

### PR DESCRIPTION
**Pull request by ERussel**
- add fee metadata request for transfer
- add fee display for transfer amount screen
- add fee calculation factory

### CODE REVIEW

File: CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+* Copyright Soramitsu Co., Ltd. All Rights Reserved.
+* SPDX-License-Identifier: GPL-3.0
+*/
+
+import Foundation
+
+public protocol FeeCalculationStrategyProtocol {
+    func calculate(for amount: Decimal) throws -> Decimal
```
Comment (unklayman) [inline CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift:9]: Why throws?

File: CommonWallet/Classes/Modules/Amount/AmountPresenter.swift [hunk 2]
```diff
@@ -8,14 +8,25 @@ import RobinHood
...
 enum AmountPresenterError: Error {
-    case missingDescriptionValidator
+    case missingSelectedAsset
 }
 
-final class AmountPresenter {
-    enum InputState {
-        case waiting
-        case checking
+struct TransferCheckingState: OptionSet {
```
Comment (unklayman) [inline CommonWallet/Classes/Modules/Amount/AmountPresenter.swift:14]: I think it should be extracted in a corresponding file.

**Review by unklayman — APPROVED** [review]

File: CommonWallet/Classes/Common/Fee/FeeCalculationFactory.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* Copyright Soramitsu Co., Ltd. All Rights Reserved.
+* SPDX-License-Identifier: GPL-3.0
+*/
+
+import Foundation
+import IrohaCommunication
+
+enum FeeType: String, Codable, Equatable {
+    case factor = "FACTOR"
+    case fixed = "FIXED"
```
Comment (unklayman) [inline CommonWallet/Classes/Common/Fee/FeeCalculationFactory.swift:11]: Have it been documented?

File: CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift (hunk 1 above)
Comment (ERussel) [reply CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift:9]: To have an opportunity to throw an error for invalid input.

File: CommonWallet/Classes/Common/Fee/FeeCalculationFactory.swift (hunk 3 above)
Comment (ERussel) [reply CommonWallet/Classes/Common/Fee/FeeCalculationFactory.swift:11]: Yes, I have added this to Capital API documentation.

File: CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift (hunk 1 above)
Comment (unklayman) [reply CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift:9]: But it's decimal in input. And the realizations don't throw as well

Comment (ERussel) [reply CommonWallet/Classes/Common/Fee/FeeCalculationStrategy.swift:9]: One can imagine cases that a value should be not less than a minimal one.

**Pull request merged**